Python/ast_unparse.c

```diff
@@ -86,7 +77,9 @@ append_repr(_PyUnicodeWriter *writer, PyObject *obj)
     if (!repr) {
         return -1;
     }
-    if ((PyFloat_CheckExact(obj) || PyComplex_CheckExact(obj))) {
+    if ((PyFloat_CheckExact(obj) && Py_IS_INFINITY(PyFloat_AS_DOUBLE(obj))) ||
+       PyComplex_CheckExact(obj))
+    {
         PyObject *new_repr = replace_inf_value(repr);
```

Comment:
You can remove the replace_inf_value() function and call directly `PyUnicode_Replace(repr, _str_inf, _str_replace_inf, -1)` here.